File: src/Client/Logistics.OfficeApp/src/app/core/api/models/trip/trip-stop.dto.ts
import {AddressDto} from "../address.dto";
import {TripStopType} from "./enums";

export interface TripStopDto {
  id: string;
  order: number;
  type: TripStopType;
  address: AddressDto;
  addressLong: number;
  addressLat: number;
  arrivedAt?: Date;
  loadId: string;
}

File: src/Shared/Logistics.Shared.Models/Trip/TripStopDto.cs
using Logistics.Domain.Primitives.Enums;
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public record TripStopDto
{
    public Guid Id { get; set; }
    public int Order { get; set; }
    public TripStopType Type { get; set; }

    public required Address Address { get; set; }
    public required GeoPoint Location { get; set; }
    public DateTime? ArrivedAt { get; set; }

    public Guid LoadId { get; set; }
    public Guid TripId { get; set; }

    public int GetDemand()
    {
        return Type is TripStopType.PickUp ? 1 : -1;
    }
}
